blog/06-02-2023-catering-to-the-client-side.md

```diff
@@ -21,21 +21,21 @@ The key distinction between the client- and server-side feature flagging is the
 
 In contrast, with a client-side app all feature flagging decisions are made in the context of the same user - the user interacting with the client-side app - and so the evaluation context is relatively static. There are cases where evaluation context will change within a client-side app - when a user logs in, for example - but by and large with client-side code we can treat feature flag evaluation context as something that is fixed (while still providing mechanisms to update it).
 
-### Flag evaluation is slow
+### The network is slow
 
-With server-side flags, we can assume that evaluating a feature flag is a _relatively_ fast operation. In some scenarios a flagging decision can be made in-process (a very fast operation). If that's not possible then flag evaluation is typically a quick service call - akin to making a DB query or calling a remote cache.
+With server-side flags, we can assume that evaluating a feature flag is a _relatively_ fast operation. With some systems the flagging rulesets[^1] live right next to where a flagging decision is needed, with flag evaluation happening either within the same process or in some sort of sidecar process. In this *local evaluation* model every flagging decision is a very fast operation. For frameworks that use a *remote evaluation* model, a flagging decision is still just a quick service call - akin to making a DB query or calling a remote cache.
 
-This situation is quite different with client-side flags. It's typically not possible to evaluate a flagging decision locally. Local evaluation would require the full feature flagging ruleset to be exposed to the client, something many organizations are not comfortable with. Additionally, flag evaluation may require additional context that's simply not available on the client at the time of evaluation.
+[^1]: by "rulesets" I mean the set of feature flagging rule configurations that define how flagging decisions should be made for each feature flag: "enable red_checkout_button flag for 50% of users", "only enable new_reco_algorithm for users in the 'internal_testers' group", etc. Rulesets plus evaluation context are the two inputs that fully define the output for any flagging decision.
```

Comment:
You may want to consider using an [Admonition](https://docusaurus.io/docs/markdown-features/admonitions) or [quote](https://docusaurus.io/docs/markdown-features#quotes).

Reply:
Ignore this... your way is better 😄